docs/DeveloperGuide.md

```diff
@@ -2,37 +2,420 @@
+corresponding `Topic`'s  `.txt`
+- updates the corresponding `Topic`'s  `.txt` whenever a `add` or
+`remove` command is called by the user
+- reads from each `Topic`'s  `.txt` and returns a `Topic` object when
+initializing the application
+
+
+### TopoSort
+**API** : `TopoCommand.java`
+
+Here is a class diagram of the `TopoCommand` which facilitates the storage
+function of the application.
+
+![](.\\uml\\diagrams\\TopoCommandClass.png "TopoCommand Class Diagram")
+
+The `TopoCommand` component
+- can topologically sort `CS2040CFile` objects in a specific `topic` order
+- can print out the list of topologically sorted `CS2040CFile` objects
+- can check whether there are `CS2040CFile` objects within `CLIAlgo` and inform user if no such objects are saved
+
+
+#### Add
+**API** : `AddCommand.java`
+
+Here is a class diagram of the `AddCommand` which is responsible for adding either code files or note files
+
+![](.\\uml\\diagrams\\Add.png "AddCommand Class Diagram")
+
+The `AddCommand` component
+- can check if the CS2040CFile to be added into our CLIAlgo exists within the same directory as the program
+- can check for the type of CS2040CFile, whether it is `.txt` or `.cpp` based on the name of the CS2040CFile
+- can ensure that there are no files with repeated names such that all names of files added are unique
+
+## Implementation
+### Filter by keyword feature
+#### Current Implementation
+The filter mechanism is facilitated by `FilterCommand`. It extends `Command` with an overridden `execute()` method. The
+`FilterCommand` has 2 different executions depending on the constructor used to instantiate it. During execution, the
+`FilterCommand` object calls either the `getNotesByTopic()` or the `getAllNotesByTopic()` methods in the `TopicManager`.
+Additionally, it implements the following operations.
+
+- `printAllTopics()` - Prints out all notes stored in CLIAlgo that is sorted by `topic`.
+- `printSingleTopic()` - Prints out all notes stored in CLIAlgo that is tagged to the given `topic`.
+
+The access modifiers of these methods are `private` can can only be accessed within `FilterCommand`.
+
+Given below is an example usage of how the filter mechanism behaves at each step.
+
+> **Step 1**: The user enters a command. The full command is read in by the `Ui` and processed by the `Parser`. If the
+> user entered a valid command, the `Parser` will process the full command using the `StringManipulation` interface and
+> prepare the appropriate `FilterCommand` object.
+
+> **Step 2**: If the `topic` field is left empty, the `Parser` will instantiate a new `FilterCommand` object, setting
+> the `topic` field to be `null`. If the `topic` field is filled with a valid topic name, the `Parser` will instantiate
+> a new `FilterCommand` using its constructor.
+
+### Initializing previous saved data feature
+#### Current implementation
+
+![](.\\sequence\\diagrams\\InitializationFileManager.png "FileManager Initialization Sequence Diagram")
+
```

Comment:
I think you might need an arrow back from `SingleFile` back to `FileManager` after the `readFile()`.